### PR TITLE
[mongoose] capped types fix

### PR DESCRIPTION
**Pull request by maxsh8x**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: http://mongoosejs.com/docs/guide.html#capped

### CODE REVIEW

File: mongoose/v3/index.d.ts [hunk 1]
```diff
@@ -312,7 +312,7 @@ declare module "mongoose" {
   export interface SchemaOption {
     autoIndex?: boolean;
     bufferCommands?: boolean;
-    capped?: boolean;
+    capped?: boolean|number|Object;
```
Comment (TonyPythoneer) [inline mongoose/v3/index.d.ts]: object is not good. Please refer this type of { size: 1024, max: 1000, autoIndexId: true }.

**Review by TonyPythoneer — CHANGES_REQUESTED** [review]
Please improve it.

**Comment by maxsh8x** [comment]
@TonyPythoneer done

File: mongoose/v3/index.d.ts [hunk 2]
```diff
@@ -309,10 +309,16 @@ declare module "mongoose" {
...
+    size: number;
+    max?: number;
+    autoIndexId?: boolean;
+  }
+
   export interface SchemaOption {
     autoIndex?: boolean;
     bufferCommands?: boolean;
-    capped?: boolean;
+    capped?: boolean|number|CappedOptions;
```
Comment (TonyPythoneer) [inline mongoose/v3/index.d.ts]: It should improve format.
```
capped?: boolean | number | CappedOptions;
```

**Review by TonyPythoneer — CHANGES_REQUESTED** [review]
Please improve format.

**Comment by maxsh8x** [comment]
@TonyPythoneer format has been improved

**Comment by TonyPythoneer** [comment]
Hello @horiuchi 
Would you review this?

**Comment by horiuchi** [comment]
@maxsh8x Thanks!
@TonyPythoneer Thank you for your reviewing!

**Pull request merged**